### ISSUE TITLE
Transfer Stake instead of burning on failed challenge

### ISSUE THREAD

**Issue by rbharath** [computablelabs/goest#133]
At present, when a listing is removed by `removeListing` in `Listing.vy`, all stake associated with that listing is burned. If the listing owner was offline during a challenge period, it is possible that they might lose all their stake associated with that listing. This feature is a vestigial remnant from earlier versions of the codebase that were based on TCRs which have a greater focus on active challenge mechanisms.

In a data market, listings may be removed or altered for reasons that have to do with data structuring. It is not necessary to punish the maker for these changes. For this reason, we propose switching to a "no-fault" listing removal mechanism. This means that upon a listing's removal, its stake is not burned. Rather it is transferred to the listing owner. In particular https://github.com/computablelabs/goest/blob/master/contracts/Listing.vy#L130, which reads
```
supply: wei_value = self.listings[hash].supply
if supply > 0:
  self.market_token.burn(supply)
```
should be altered to
```
supply: wei_value = self.listings[hash].supply
if supply > 0:
  self.market_token.transfer(self.listings[hash].owner, supply)
```

**Comment by rbharath** [comment]
Fixed by #146

**Comment by rbharath** [comment]
Re-opening after rollback

**Comment by rbharath** [comment]
Re-implemented by #169
